### ISSUE TITLE
'OutlineExplorerTreeWidget' object has no attribute 'update_all_editors'

### ISSUE THREAD

**Issue by sphh** [spyder-ide/spyder#16320]
### What steps will reproduce the problem?

I believe this happens, whenever I switch back to the Spyder window, whenever I have been away for a long time / the computer has hibernated.

I have the following view open:
- Editor pane split horizontally into two editors.
- Outline pane.

### Traceback
```python-traceback
Traceback (most recent call last):
  File "/usr/local/lib/python3.8/dist-packages/spyder/plugins/outlineexplorer/widgets.py", line 1013, in change_visibility
    self._change_treewidget_visibility(self.is_visible)
  File "/usr/local/lib/python3.8/dist-packages/spyder/plugins/outlineexplorer/widgets.py", line 1044, in _change_treewidget_visibility
    self.treewidget.update_all_editors()
AttributeError: 'OutlineExplorerTreeWidget' object has no attribute 'update_all_editors'
```

## Versions

* Spyder version: 5.1.1 
* Python version: 3.8.10
* Qt version: 5.12.9
* PyQt5 version: 5.12.3
* Operating System: Linux Mint 20.2 Cinnamon

### Dependencies

```

# Mandatory:
atomicwrites >=1.2.0          :  1.4.0 (OK)
chardet >=2.0.0               :  4.0.0 (OK)
cloudpickle >=0.5.0           :  1.6.0 (OK)
cookiecutter >=1.6.0          :  1.7.3 (OK)
diff_match_patch >=20181111   :  20200713 (OK)
intervaltree >=3.0.2          :  3.1.0 (OK)
IPython >=7.6.0               :  7.26.0 (OK)
jedi >=0.17.2;<0.19.0         :  0.17.2 (OK)
jsonschema >=3.2.0            :  3.2.0 (OK)
keyring >=17.0.0              :  23.0.1 (OK)
nbconvert >=4.0               :  6.1.0 (OK)
numpydoc >=0.6.0              :  1.1.0 (OK)
parso >=0.7.0;<0.9.0          :  0.7.0 (OK)
pexpect >=4.4.0               :  4.6.0 (OK)
pickleshare >=0.4             :  0.7.5 (OK)
psutil >=5.3                  :  5.5.1 (OK)
pygments >=2.0                :  2.9.0 (OK)
pylint >=2.5.0                :  2.9.6 (OK)
pyls_spyder >=0.4.0           :  0.4.0 (OK)
pylsp >=1.2.1;<1.3.0          :  1.2.1 (OK)
pylsp_black >=1.0.0           :  None (OK)
qdarkstyle =3.0.2             :  3.0.2 (OK)
qstylizer >=0.1.10            :  0.2.1 (OK)
qtawesome >=1.0.2             :  1.0.3 (OK)
qtconsole >=5.1.0             :  5.1.1 (OK)
qtpy >=1.5.0                  :  1.9.0 (OK)
rtree >=0.9.7                 :  0.9.7 (OK)
setuptools >=49.6.0           :  57.4.0 (OK)
sphinx >=0.6.6                :  4.1.2 (OK)
spyder_kernels >=2.1.0;<2.2.0 :  2.1.0 (OK)
textdistance >=4.2.0          :  4.2.1 (OK)
three_merge >=0.1.1           :  0.1.1 (OK)
watchdog >=0.10.3             :  1.0.2 (OK)
xdg >=0.26                    :  0.26 (OK)
zmq >=17                      :  22.2.0 (OK)

# Optional:
cython >=0.21                 :  None (OK)
matplotlib >=2.0.0            :  3.4.2 (OK)
numpy >=1.7                   :  1.21.1 (OK)
pandas >=1.1.1                :  1.3.1 (OK)
scipy >=0.17.0                :  1.7.1 (OK)
sympy >=0.7.3                 :  1.8 (OK)
```

**Comment by sphh** [comment]
Here is a way, how I can reliably reproduce this error message: I have (virtual) workspaces defined. Whenever I switch back from another workspace to that one with Spyder on it, the above exception is raise.

**Comment by ccordoba12** [comment]
Hey @sphh, thanks for reporting. This is not a problem with Spyder but (most probably) because you're mixing patches from our 5.x branch with the current code in 5.1.1.

I say this because I just performed a search in the code part of 5.1.1 and there's no single reference to `update_all_editors` (I removed for that version). However, I re-added it in 5.x to solve some issues.

Due to that, I'm going to close this issue because it's not a real bug.
